models/comments-model.js

```diff
@@ -1,11 +1,34 @@
-const db = require('../db/connection');
+const db = require("../db/connection");
 
 exports.selectComments = (article_id) => {
-    let queryStr = `
+  let queryStr = `
     SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`;
 
-    return db.query(queryStr, [article_id]).then(({ rows }) => {
-        const comments = rows;
-        return comments;
-    })
-}
\ No newline at end of file
+  return db.query(queryStr, [article_id]).then(({ rows }) => {
+    return rows;
+  });
+};
+
+exports.postComment = ( username, body, article_id) => {
+  if (!username && !body) {
+    return Promise.reject({
+      status: 400,
+      msg: "Bad request",
+    });
+  }
+
+  if (!body.length || !username.length) {
+    return Promise.reject({
+      status: 400,
+      msg: "Please enter a comment before posting",
+    });
+  }
+
+  let queryStr = `
+    INSERT INTO comments(author, body, article_id) VALUES ($1, $2, $3) RETURNING *`;
+
+  return db.query(queryStr, [username, body, article_id]).then(({ rows }) => {
+    const comment = rows[0];
+    return comment;
+  });
+};
```

Comment:
Some good customer error handling here but also you've demonstrated a good understanding of which errors Postgres will throw.